### DISCUSSION TITLE
How reactive({}) is a possible vuln on SSR?

### DISCUSSION THREAD

**Discussion by AvdylKrasniqi** [vuejs/pinia#921] [Help and Questions] [answered]
Accoring to  pinia home page

`If you are familiar with the Composition API, you might be thinking you can already share a global state with a simple export const state = reactive({}). This is true for single page applications but exposes your application to security vulnerabilities if it is server side rendered. 
`

Can you please explain how reactive is a possible security vulnerability when it is SSR? Any real world example, article or something else as I couldn't find anything on google!

**Comment by posva** [comment, marked as answer]
https://ssr.vuejs.org/guide/structure.html#avoid-stateful-singletons

**Reply by QzCurious** [reply]
The provided link is not a valid for now. I thought it is great to link to a new one.

https://vuejs.org/guide/scaling-up/ssr.html#cross-request-state-pollution
